### PR TITLE
fix various type and function signatures for RIOT

### PR DESCRIPTION
**Pull request by cgundogan**
### Contribution description
I encountered several compiler errors due to recent refactorings when building the current master for RIOT.

### Issues/PRs references
--

### CODE REVIEW

File: src/ccnl-fwd/src/ccnl-fwd.c [hunk 1]
```diff
@@ -122,7 +122,7 @@ ccnl_fwd_handleContent(struct ccnl_relay_s *relay, struct ccnl_face_s *from,
 #ifdef USE_RONR
     /* if we receive a chunk, we assume more chunks of this content may be
      * retrieved along the same path */
-    if ((c->pkt->pfx->chunknum) && (*(c->pkt->pfx->chunknum) >= 0)) {
+    if (c->pkt->pfx->chunknum) {
```
Comment (mfrey) [inline src/ccnl-fwd/src/ccnl-fwd.c]: Why did you remove the``c->pkt->pfx->chunknum) >= 0)`` part? First check if the pointer is valid and second if there is actually more than a single chunk, or?

Comment (cgundogan) [reply src/ccnl-fwd/src/ccnl-fwd.c]: the compiler was complaining, because a pointer is never `< 0`.

Comment (mfrey) [reply src/ccnl-fwd/src/ccnl-fwd.c]: 👍
